### PR TITLE
Fix VDI migration on the right SR

### PR DESCRIPTION
**Pull request by Wescoeur**

### CODE REVIEW

**Review by gduperrey — APPROVED** [review]

File: lib/basevm.py [hunk 1]
```diff
@@ -63,6 +63,12 @@ def all_vdis_on_host(self, host):
                 return False
         return True
 
+    def all_vdis_on_sr(self, sr):
```
Comment (stormi) [inline lib/basevm.py:66]: Do we still need the all_vdis_on_host method? If yes, should we make it return False for shared SRs?

File: tests/migration/test_host_evacuate.py [hunk 2]
```diff
@@ -20,8 +20,7 @@
 #   This network will be disconnected at some point during the tests
 
 def _host_evacuate_test(source_host, dest_host, network_uuid, vm, expect_error=False, error=""):
-    source_name = source_host.xe('host-param-get', {'uuid': source_host.uuid, 'param-name': 'name-label'})
-    vm.start(on=source_name)
+    vm.start(on=source_host.uuid)
```
Comment (stormi) [inline tests/migration/test_host_evacuate.py:23]: IIRC there's also a param that must be removed from above the vm.start method definition.

File: tests/storage/moosefs/conftest.py [hunk 3]
```diff
@@ -13,15 +14,24 @@ def host_with_moosefs(host):
...
+def pool_with_moosefs(hostA1):
+    pool = Pool(hostA1.hostname_or_ip)
+    for host in pool.hosts:
+        enable_moosefs(host)
+
+    yield pool
+
+    for host in pool.hosts:
+        disable_moosefs(host)
+
```
Comment (stormi) [inline tests/storage/moosefs/conftest.py:34]: This fixture will behave poorly if for example the third host has moosefs installed on it for some reason. But it's not worse than in other similar tests, so I think I'll try to fix them all at some point.

Comment (stormi) [reply tests/storage/moosefs/conftest.py:34]: test_reboot should also probably explicitly start the VM on the master host as this is the only host that was rebooted.

**Review by stormi — APPROVED** [review]

**Pull request merged**